core/templates/pages/splash-page/splash-page.component.ts

```diff
@@ -30,45 +30,45 @@ import splashConstants from 'assets/constants';
 angular.module('oppia').component('splashPage', {
   template: require('./splash-page.component.html'),
   controller: [
-    '$rootScope', 'LoaderService', 'SiteAnalyticsService',
+    '$rootScope', '$translate', 'LoaderService', 'SiteAnalyticsService',
     'UrlInterpolationService', 'UserService', 'WindowDimensionsService',
     function(
-        $rootScope, LoaderService, SiteAnalyticsService,
+        $rootScope, $translate, LoaderService, SiteAnalyticsService,
         UrlInterpolationService, UserService, WindowDimensionsService) {
       var ctrl = this;
       ctrl.getStaticImageUrl = function(imagePath) {
-        return UrlInterpolationService.getStaticImageUrl(imagePath);
+        if (imagePath) {
+          return UrlInterpolationService.getStaticImageUrl(imagePath);
+        }
       };
 
       ctrl.onClickBrowseLessonsButton = function() {
         SiteAnalyticsService.registerClickBrowseLessonsButtonEvent();
         return false;
       };
 
-      ctrl.getMainBackgroundUrl = function() {
-        if (ctrl.isWindowNarrow()) {
-          return ctrl.getStaticImageUrl('/splash/splashMainMobile.png');
-        } else {
-          return ctrl.getStaticImageUrl('/splash/splashMainDesktop.png');
-        }
-      };
-
       ctrl.isWindowNarrow = function() {
         return WindowDimensionsService.isWindowNarrow();
       };
 
-      // The 2 functions below are to cycle between values: 0, 1, 2 for
-      // testimonialId.
-      ctrl.incrementTestimonialId = function() {
-        // This makes sure that incrementing from 2, returns 0 instead of 3,
-        // since we want the testimonials to cycle through.
-        ctrl.testimonialId = (ctrl.testimonialId + 1) % 3;
+      // The 2 functions below are to cycle between values:
+      // 0 to testimonialCount - 1 for displayedTestimonialId.
+      ctrl.incrementDisplayedTestimonialId = function() {
+        // This makes sure that incrementing from testimonialCount - 1, returns
+        // 0 instead of testimonialCount,since we want the testimonials to cycle
+        // through.
+        ctrl.displayedTestimonialId = (
+          ctrl.displayedTestimonialId + 1) % ctrl.testimonialCount;
       };
 
-      ctrl.decrementTestimonialId = function() {
-        // This makes sure that decrementing from 0, returns 2 instead of -1,
-        // since we want the testimonials to cycle through.
-        ctrl.testimonialId = (((ctrl.testimonialId - 1) % 3) + 3) % 3;
+      ctrl.decrementDisplayedTestimonialId = function() {
+        // This makes sure that decrementing from 0, returns
+        // testimonialCount - 1 instead of -1, since we want the testimonials to
+        // cycle through.
+        ctrl.displayedTestimonialId = (
+          ((ctrl.displayedTestimonialId - 1) % ctrl.testimonialCount) +
```

Comment:
Typically, just do:

    (ctrl.displayedTestimonialId + ctrl.testimonialCount - 1) % ctrl.testimonialCount

and then you save on one % operation.

Reply:
Done